packages/core/src/navigation/navigation.element.ts

```diff
@@ -0,0 +1,479 @@
+      }
+    });
+
+    // Is it an item inside of a group?
+    // is the group expanded? -> move focus to the group start focusable element
+    onKey('arrow-left', e, () => {
+      // is this a start element inside of a group?
+      // is the group expanded?
+      // => emit the group expandedChange event
+      if (isGroupStartEvent(e) && expandedGroupEvent(e)) {
+        fireGroupExpandedChangeEvent(e);
+      }
+
+      if (this.currentActiveItem && isGroupItemEvent(e)) {
+        // remove focus from currently focused element
+        // focus on group start element
+        removeFocus(this.currentActiveItem);
+        const startGroupParent = getStartGroupElement(e) as CdsNavigationStart;
+        this.ariaActiveDescendant = startGroupParent.id;
+        setFocus(startGroupParent as FocusableElement);
+      }
+
+      if (isRootStart(e) && expandedNavigation(e)) {
+        this.toggle();
+      }
+    });
+
+    onKey('home', e, () => {
+      if (this.currentActiveItem) {
+        removeFocus(this.currentActiveItem);
+        const home = focusableElements[0];
+        this.ariaActiveDescendant = home.id;
+        setFocus(home);
+      }
+    });
+
+    onKey('end', e, () => {
+      if (this.currentActiveItem) {
+        removeFocus(this.currentActiveItem);
+        const end = focusableElements[focusableElements.length - 1];
+        this.ariaActiveDescendant = end.id;
+        setFocus(end);
+      }
+    });
+  };
+
+  private toggle() {
+    this.expandedChange.emit(!this.expanded);
+  }
+
+  get currentActiveItem() {
+    return this.visibleChildren.find(c => c.id === this.ariaActiveDescendant);
+  }
+
+  protected get footerTemplate() {
+    return this.navigationEnd
+      ? html`
+          <div role="region" class="navigation-end">
+            <slot name="cds-navigation-end"></slot>
+          </div>
+        `
+      : '';
+  }
+
+  protected get startTemplate() {
+    let returnHTML;
+
+    this.rootNavigationStart
+      ? (returnHTML = html`
+          <div role="region" class="navigation-start" cds-layout="vertical">
+            <slot name="navigation-start"></slot>
+            <cds-divider class="start-divider"></cds-divider>
+          </div>
+        `)
+      : (returnHTML = '');
+
+    return returnHTML;
+  }
+
+  protected get visibleChildren(): FocusableElement[] {
+    return Array.from(this.allNavigationElements).filter(n => isVisible(n));
+  }
+
+  firstUpdated() {
+    if (this.rootNavigationStart) {
+      this.rootNavigationStart.addEventListener('click', () => {
+        this.toggle();
+      });
+    }
+
+    // set all visible navigation elements to tabindex = -1
+    this.allNavigationElements.forEach(item => {
+      setAttributes(item, ['tabindex', '-1']);
+    });
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener('focus', this.initAriaActiveDescendant);
+    this.removeEventListener('keydown', this.keyboardNavigationHandler);
```

Comment:
It seems we are removing two out of four event listeners. Is that intentional?

Reply:
Nope. I updated this and we are now removing five event listeners in disconnectedCallback.